### DISCUSSION TITLE
How to add custom context menu in Region plugin

### DISCUSSION THREAD

**Discussion by XyzAQWERTY** [katspaugh/wavesurfer.js#3092] [Q&A] [answered]
I want to add custom context menu when right click in the region plugin. i.e. I want to add delete button on a region, how do I achieve this?

**Comment by katspaugh** [comment, marked as answer]
You can pass any HTML element in the `content` property of a region. E.g.

```js
Regions.addRegion({
    start: 0,
    end: 8,
    content: contextMenuElement,
  })
```

**Comment by XyzAQWERTY** [comment]
I can pass HTML element using `document.createElement("")`. But can I use react component for the content property?

**Reply by katspaugh** [reply]
No, you need to use a ref in React.

**Reply by XyzAQWERTY** [reply]
Okay, Thanks
